tools/alive_parser.cpp

```diff
@@ -418,10 +446,18 @@ static unique_ptr<Instr> parse_binop(string_view name, token op_token) {
   case UADD_SAT: op = BinOp::UAdd_Sat; break;
   case SSUB_SAT: op = BinOp::SSub_Sat; break;
   case USUB_SAT: op = BinOp::USub_Sat; break;
+  case SADD_OVERFLOW:
+    op = BinOp::SAdd_Overflow;
+    rettype = std::ref(get_overflow_type(type.bits()));
+    break;
+  case EXTRACTVALUE:
+    op = BinOp::ExtractValue;
+    rettype = std::ref(get_sym_type());
```

Comment:
why override rettype?

Reply:
because `extractvalue` is different than other instructions in the sense that it doesn't return the same type as its first argument - it rather returns type based on second argument; therefore, I assigned it a symbolic type to be determined later.

Reply:
ok, makes sense since we aren't parsing the result type, but please get rid of this std::ref ugliness :)

Reply:
Just wanted to avoid raw pointers; thought reference_wrapper is cleaner than using raw pointers. But okay, i have removed the use of reference wrapper now.